validation-test/stdlib/String.swift

```diff
@@ -999,6 +1000,30 @@ StringTests.test(
 #endif
 }
 
+StringTests.test("String._fastCStringContents()")
+  .skip(.nativeRuntime("_NSContiguousString undefined without _runtime(_ObjC)"))
+  .code {
+#if _runtime(_ObjC)
+  // ASCII tests
+  for i in (0..<128) {
+    let cs = _NSContiguousString(String(UnicodeScalar(i)!)._core)
+    expectNotEqual(cs._fastCStringContents(), nil)
+  }
```

Comment:
Could you also add tests for slices?  Since slices share their data buffer with the complete string, it is important that we return `nil` for slices that hide the tail of the character sequence.